packages/core/src/common/_flex.scss

```diff
@@ -0,0 +1,47 @@
+// Copyright 2018 Palantir Technologies, Inc. All rights reserved.
+// Licensed under the terms of the LICENSE file distributed with this project.
+
+// this element becomes a flex container in the given direction.
+// supply `$margin` to put space between each child.
+// supply `$inline: inline` to use `display: flex-inline`.
+// customize `flex: 1 1` child selector with $fill.
+@mixin pt-flex-container($direction: row, $margin: none, $inline: none, $fill: ".pt-fill") {
+  @if $inline == inline or $inline == true {
+    display: inline-flex;
+  } @else {
+    display: flex;
+  }
+  flex-direction: $direction;
+
+  > * {
+    flex: 0 0 auto;
+  }
+
+  > #{$fill} {
+    flex: 1 1 auto;
+  }
+
+  @if $margin != none {
+    @include pt-flex-margin($direction, $margin);
+  }
+}
+
+// applies margin along axis of direction between every direct child, with no margins on either end.
+// $direction: row | column
+// $margin: margin[-right|-bottom] value
+@mixin pt-flex-margin($direction, $margin) {
```

Comment:
this whole thing inspired by "Spacing children" section on https://typestyle.github.io/#/box.
I used a similar pattern to great success in recent app development: huge reduction in layout headaches when you apply spacing to all children at the container level.

Reply:
super neat. since we don't export Sass functions/mixins, what do you think about providing utility classes?

Reply:
we do expose these sass functions... we ship all the original .scss files.
i'd be open to adding classes, but that can be done additively later. it's tricky cuz the mixin takes a bunch of parameters, but classes are static styles so we'd have to choose defaults (less flexible) or provide many classes (unwieldy).